services/ext/api.go

```diff
@@ -581,6 +582,27 @@ func (api *PublicAPI) CanceledRequestsToJoinForCommunity(id types.HexBytes) ([]*
 	return api.service.messenger.CanceledRequestsToJoinForCommunity(id)
 }
 
+// Generates a single hash for each address that needs to be revealed to a community.
+// Each hash needs to be signed.
+// The order of retuned hashes corresponds to the order of addresses in addressesToReveal.
+func (api *PublicAPI) GenerateJoiningCommunityRequestsForSigning(memberPubKey string, communityID types.HexBytes, addressesToReveal []string) ([]account.SignParams, error) {
```

Comment:
Why explicit `memberPubKey` is needed here? Can it be different than our own?

Reply:
At the moment no, cause the user who is joining a community is the logged-in user. But this function generates joining community requests based on passed parameters, so it's general.

Reply:
I think it's very unlikely that we'll need to generate requests for accounts other than our own. Perhaps considering the YAGNI principle here might help in keeping the API more straightforward?